### PR TITLE
Backlight and RGB now working for JJ50

### PR DESCRIPTION
**Pull request by WarmCatUK**
Backlight / RGB now working for JJ50, code tidied up somewhat.

### CODE REVIEW

File: keyboards/ymd96/ymd96.h [hunk 1]
```diff
@@ -41,6 +41,19 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 	{      K700,       K701,       K702,       K703,       K704,       K705,       K706,       KC_NO,      K708,       K709,       K710,       K711,       K712,       K713,      K714  } \
 }
 
+#define KEYMAP_JJ50( \
```
Comment (mechmerlin) [inline keyboards/ymd96/ymd96.h]: Could you rename `KEYMAP_JJ50` to `LAYOUT_jj50`to adhere to the new QMK standards.

File: keyboards/jj50/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+     * |      |      |      |      |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |      |      |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |BL_OFF|BL_DEC|BL_INC|BL_ON |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_FN] = KEYMAP( \
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/default/keymap.c]: Could you rename this to `LAYOUT`

File: keyboards/jj50/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+     * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Del  |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |   [  |   ]  |  \   |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |ISO # |ISO / |  PUP |  PDN |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_RAISE] = KEYMAP( \
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/default/keymap.c]: Could you rename this to `LAYOUT`

File: keyboards/jj50/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+     * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter | (Hold for shift / tap for enter)
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * | _FN  | Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_DEFLT] = KEYMAP( \
```
Comment (mechmerlin) [inline keyboards/jj50/keymaps/default/keymap.c]: Could you rename this to `LAYOUT`

**Comment by WarmCatUK** [comment]
Yep, done, I'll create a new merge request with the changes.

**Comment by WarmCatUK** [comment]
Yes, it was me who originally submitted jj50 months ago, and I've amended the code considerably in the process of getting backlighting working; hence conflicts.

**Comment by drashna** [comment]
@jackhumbert somehow, Travis didn't even show up here....

**Comment by drashna** [comment]
Ah, well, the conflicts need to be resolved, properly, or else we won't be able to merge it, actually.  

And while we could do this ourselves, it would be best to allow you to do this, since you either have a board to test with, or are much more familiar with the design. Or both.

**Comment by WarmCatUK** [comment]
I am happy to resolve conflicts, if you point me in the right direction; I am not fluent in GitHub.

**Comment by WarmCatUK** [comment]
Is it best that I close this pull request and submit a new one? Then perhaps Travis will resolve conflicts automatically.

**Comment by drashna** [comment]
Rebasing may be the simplest way to handle this.

**Comment by WarmCatUK** [comment]
Ok, just reading up on how to rebase, looks rather complicated (oh my god I hate GitHub), so bear with me.

**Comment by drashna** [comment]
hahaha, yeah.  Merging may work too, but .... rebasing it better in the long run.  

But if you have merge conflicts (you will), then it is a PITA.   I've seen a couple people recommend GitKraken, I think.

**Comment by jackhumbert** [comment]
We'll end up squashing the commits, so you can merge if that's easier. Resolving the conflicts can be done on the commandline by pulling, but it's probably easier on Github via the "Resolve Conflicts" button just below the comments here - you should have access to this since it's your PR.

**Comment by WarmCatUK** [comment]
Thanks for the suggestion, but I don't have access :-/
This is frustrating as I don't often use GitHub. Perhaps it's easier for me to delete my fork and create a new one, add my amended files and then create a new pull request.
Such a simple thing is a total nightmare in GitHub.

**Comment by WarmCatUK** [comment]
Heh, additionally I have just ported to YMDK's YMD75 pcb, so I have that to submit after this one :-)

**Comment by drashna** [comment]
You should be able to merge from upstream and then push to your branch and fork on github.  That will update the PR automatically (eg, no need to close and reopen)

**Comment by drashna** [comment]
Wooo, that worked. :)

Thanks!

**Comment by WarmCatUK** [comment]
Hah! Thanks for your help! I'll make sure my fork is fully up-to-date before making any future PRs!

**Pull request merged**